src/adagrams.js

```diff
@@ -1,15 +1,140 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+"use strict";
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
 };
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
+const SCORE_CHART = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
 };
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
-};
+class Adagrams {
+  drawLetters() {
+    // Implement this method for wave 1
+    let newPool = [];
+    for (const [letter, count] of Object.entries(LETTER_POOL)) {
+      for (let i = 0; i < count; i++) {
+        newPool.push(letter);
+      }
+    }
+
+    let letters = [];
+    while (letters.length < 10) {
+      let draw = newPool[Math.floor(Math.random() * newPool.length)];
+
+      let letter = newPool.pop(draw);
+      letters.push(letter);
+    }
+    return letters;
+  }
+
+  usesAvailableLetters(input, lettersInHand) {
+    // Implement this method for wave 2
+    let uppercaseWord = input.toUpperCase();
+    let lettersInHandCopy = lettersInHand.slice();
+
+    for (let char of uppercaseWord) {
```

Comment:
Since we're not reassigning `char` in the loop, could we use const here instead?